lib/ui/painting/canvas.cc

```diff
@@ -2,10 +2,11 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#define _USE_MATH_DEFINES
```

Comment:
Why not keep this with the header?

(Applies to the same change in other files as well.)

Reply:
In a bunch of cases, we pull in cmath without defining this; if the first include is via something transitively included via canvas.h, we end up getting it without `M_PI` etc.

There are a couple options: one is that everywhere we include cmath, we define this. Another is the approach of defining this symbol before the first include in a given file, which feels safer, despite the fact it would be nice to keep them side-by-side. Thoughts/preferences?

Reply:
Could we add it as a define in GN instead?

Reply:
That's a good point. Will take a look; that avoids cluttering the code with this altogether. It's not obvious to readers at first blush why this is there until you look up what the define does and it's not anything insightful when you do finally find out.

Reply:
Done in https://github.com/flutter/engine/pull/21166